server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateService.java

```
@@ -527,6 +552,18 @@ private UploadedMetadataResults writeMetadataInParallel(
             uploadTasks.put(indexMetadata.getIndex().getName(), getIndexMetadataAsyncAction(clusterState, indexMetadata, listener));
         });
 
+        indicesRoutingToUpload.forEach(indexRoutingTable -> {
+            try {
+                uploadTasks.put(
+                    indexRoutingTable.getIndex().getName() + "--indexRouting",
+                    remoteRoutingTableService.get().getIndexRoutingAsyncAction(clusterState, indexRoutingTable, listener, getCusterMetadataBasePath(clusterState.getClusterName().value(),
+                        clusterState.metadata().clusterUUID()))
+                );
+            } catch (IOException e) {
+                e.printStackTrace();
```

lets throw the exception